### PR TITLE
[FLINK-26436] InformerEventSources should only watch selected namespace

### PR DESCRIPTION
**Pull request by gyfora**

### CODE REVIEW

**Comment by SteNicholas** [comment]
@gyfora , in general, the pull request LGTM. Could you please add the unit tests for `OperatorUtils`?

**Comment by gyfora** [comment]
Thanks, will add some tests today

File: flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentControllerTest.java [hunk 1]
```diff
@@ -52,10 +62,29 @@
...
+        flinkService = new TestingFlinkService();
+        mockServer = new KubernetesMockServer(new MockWebServer(), new HashMap<>(), true);
+        mockServer.init();
+        kubernetesClient = mockServer.createClient();
+        testController = createTestController(kubernetesClient, flinkService);
+    }
+
+    @AfterEach
+    public void tearDown() {
+        mockServer.shutdown();
```
Comment (SteNicholas) [inline flink-kubernetes-operator/src/test/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentControllerTest.java:82]: Could the mock server shutdown after the kubernetes client close?

**Comment by SteNicholas** [comment]
@gyfora, thanks for your updates of the tests. There are something else conflicts here and please resolve the conflicts.

**Review by SteNicholas — APPROVED** [review]
LGTM. cc @wangyang0918 @tweise

File: flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java [hunk 2]
```diff
@@ -156,22 +156,15 @@ private void updateForReconciliationError(FlinkDeployment flinkApp, String err)
...
-                        .withLabel(Constants.LABEL_TYPE_KEY, Constants.LABEL_TYPE_NATIVE_TYPE)
-                        .withLabel(
-                                Constants.LABEL_COMPONENT_KEY,
-                                Constants.LABEL_COMPONENT_JOB_MANAGER)
-                        .runnableInformer(0);
-        return List.of(
-                new InformerEventSource<>(
-                        deploymentInformer, Mappers.fromLabel(Constants.LABEL_APP_KEY)));
+    public List<EventSource> prepareEventSources(EventSourceContext<FlinkDeployment> ctx) {
+        Set<String> effectiveNamespaces = controllerConfig.getEffectiveNamespaces();
```
Comment (wangyang0918) [inline flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java:160]: We might need a mechanism to guard `controllerConfig` is not null.

**Review by wangyang0918 — APPROVED** [review]
Nice work. I left two minor comments.

**Comment by wangyang0918** [comment]
> I fixed the comments @wangyang0918 . Btw this feature will not be required in the future, as the operator SDK community is adding built in support for this as we speak. It will be contained in the upcoming version of the operator sdk I hope.

This is good news and maybe we could create a ticket to ensure it is not forgotten.

+1 for merging.

**Pull request merged**